subworkflow/local/mapping.nf

```diff
@@ -50,18 +50,18 @@ workflow MAPPING {
         bam_bwamem2 = Channel.empty()
 
         if (params.aligner == "bwa-mem") {
-            BWAMEM1_MEM(reads_input_status.normal, bwa, fasta, fai)
+            BWAMEM1_MEM(reads_input_status.normal, bwa)
             bam_bwamem1_n = BWAMEM1_MEM.out.bam
 
-            BWAMEM1_MEM_T(reads_input_status.tumor, bwa, fasta, fai)
+            BWAMEM1_MEM_T(reads_input_status.tumor, bwa)
             bam_bwamem1_t = BWAMEM1_MEM_T.out.bam
 
             bam_bwamem1 = bam_bwamem1_n.mix(bam_bwamem1_t)
         } else {
-            BWAMEM2_MEM(reads_input_status.normal, bwa, fasta, fai)
+            BWAMEM2_MEM(reads_input_status.normal, bwa)
             bam_bwamem2_n = BWAMEM2_MEM.out.bam
 
-            BWAMEM2_MEM_T(reads_input_status.tumor, bwa, fasta, fai)
+            BWAMEM2_MEM_T(reads_input_status.tumor, bwa)
```

Comment:
how to resolve the groupTuple block problems in line 78,   because I can not comment on that line, so I comment here, 
if mapping step has multiple run result, groupTuple woll block all bam files, until all samples have the align result, however, we can not set size paramter in groupTuple, because we don't know how many runs in one sample, there may be 2,3,4,.... and so on

Reply:
I don't see a problem here actually.
the `reads_input.groupTuple(by: [0,1])` ensure that we have only one fastq pairs in the channel emission.
We group the different bams from the same samples with this:
https://github.com/nf-core/sarek/blob/a8958e590c05fd5d59fbc647c4dd0a6a8e9585be/subworkflow/local/mapping.nf#L112-L113

Reply:
But maybe I misunderstood your comment.
Can you elaborate more if it was the case?

Reply:
What I mean is in[ this part of groupTuple](https://github.com/nf-core/sarek/blob/a8958e590c05fd5d59fbc647c4dd0a6a8e9585be/subworkflow/local/mapping.nf#L72-L82) ,  because you need to group the output bam according to bam files size, `bam_bwa_to_sort ` channel will not emit element until all aligned bam has finished. what's more, there are not set 'size' paramter in groupTuple operator

Reply:
ok, I'll do some more tests for that.
I copied part of this code from dsl1 and tried to update it, it's definitively possible that I created issues.
Thanks for noticing that

Reply:
Looking forward to your reply as soon as possible, I have come across the same problem in my code, but I don't know how to solve this problem, because the size numbers are  Dynamically changing values for different samples which multiple run data.

Reply:
Ok, re-read and tested things, and I understand what you want.
Sorry, I was still in a no coffee mode this morning.

Reply:
I'm creating a new issue, and will work on that separately, I have some idea I'd like to try to take care of this issue:
https://github.com/nf-core/sarek/issues/362